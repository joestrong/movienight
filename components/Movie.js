import React from 'react'
import { ImageBackground, Text, Image, StyleSheet } from 'react-native';

export default class Movie extends React.Component {
  render() {
    const movie = this.props.movie;
    
    return (
      <ImageBackground source={{uri:movie.backdropImage}} style={styles.movie}>
        <Image source={{uri:movie.posterImage}} style={styles.image}></Image>
        <Text style={styles.title}>{movie.title}</Text>
      </ImageBackground>
    )
  }
}

const styles = StyleSheet.create({
  movie: {
    flexDirection: 'row',
    margin: 15,
    padding: 15
  },
  image: {
    width: 150,
    height: 225
  },
  title: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    color: '#fff',
    backgroundColor: 'rgba(0,0,0,0.5)',
    fontSize: 20,
    padding: 15
  }
});
