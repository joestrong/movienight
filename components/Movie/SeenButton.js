import React from 'react'
import { Button } from 'react-native';
import MovieService from '../../services/MovieService'

export default class SeenButton extends React.Component {
  render() {
    const seenText = (this.props.movie.seen === true) ? 'Seen' : 'Unseen'

    return (
      <Button title={seenText} onPress={e => this.onPress()} />
    )
  }

  onPress(e) {
    if (this.props.movie.seen === true) {
      MovieService.removeSeen(this.props.movie)
    } else {
      MovieService.markSeen(this.props.movie)
    }
  }
}
