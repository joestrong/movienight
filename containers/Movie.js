import { connect } from 'react-redux'
import MovieComponent from '../components/Movie'

const mapStateToProps = state => {
  return {
  }
}

const mapDispatchToProps = dispatch => {
  return {
  }
}

const Movie = connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieComponent)

export default Movie
