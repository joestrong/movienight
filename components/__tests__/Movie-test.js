import 'react-native'
import React from 'react'
import Movie from '../Movie'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const movie = {
    "title": "Avengers: Endgame",
    "posterImage": "https:\/\/image.tmdb.org\/t\/p\/w500\/or06FN3Dka5tukK1e9sl16pB3iy.jpg",
    "backdropImage": "https:\/\/image.tmdb.org\/t\/p\/w500\/7RyHsO4yDXtBv1zUU3mTpHeQ0d5.jpg"
  }
  const tree = renderer.create(<Movie movie={movie}></Movie>).toJSON();

  expect(tree).toMatchSnapshot();
});
