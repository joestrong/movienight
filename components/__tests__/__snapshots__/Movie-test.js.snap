// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flexDirection": "row",
      "margin": 15,
      "padding": 15,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w500/7RyHsO4yDXtBv1zUU3mTpHeQ0d5.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <Image
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w500/or06FN3Dka5tukK1e9sl16pB3iy.jpg",
      }
    }
    style={
      Object {
        "height": 225,
        "width": 150,
      }
    }
  />
  <View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Unseen
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "bottom": 0,
        "color": "#fff",
        "fontSize": 20,
        "left": 0,
        "padding": 15,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    Avengers: Endgame
  </Text>
</View>
`;
